pkg/controllers/dashboard/systemcharts/controller.go

```diff
@@ -101,7 +111,12 @@ func (h *handler) onRepo(key string, repo *catalog.ClusterRepo) (*catalog.Cluste
 		}
 		// webhook needs to be able to adopt the MutatingWebhookConfiguration which originally wasn't a part of the
 		// chart definition, but is now part of the chart definition
-		if err := h.manager.Ensure(chartDef.ReleaseNamespace, chartDef.ChartName, chartDef.MinVersionSetting.Get(), values, chartDef.ChartName == webhookChartName, installImageOverride); err != nil {
+		minVersion := chartDef.MinVersionSetting.Get()
+		exactVersion := chartDef.ExactVersionSetting.Get()
```

Comment:
I think I get what you are going for here, but I think that this is a bit error prone for the future and could lead to nil pointers if someone has specified one of these values but not the other. I would recommend putting in a side function, something like the following:

```golang

// getChartVersion gets the min version and exact version (in that order) for a chart
def getChartVersions(chartDef chart.Definition) (string, string){
    minVersion := ""
    if chartDef.MinVersionSetting != nil{
        minVersion = chartDef.MinVersionSettting.Get()
    }
    if chartDef.ExactVersionSetting != nil{
        exactVersion = chartDef.ExactVersionSetting.Get()
    }
    return minVersion, exactVersion

}
```

Reply:
I don't foresee nil pointers here, since `chart.Definition` stores a **value** of type `Setting`, not a pointer to it.
Also, I do have tests that set one value but not the other, and there are no nil pointers. Integration tests also have such a scenario.
https://github.com/rancher/rancher/blob/release/v2.7/pkg/controllers/dashboard/systemcharts/controller_test.go

Reply:
Fair comment. As long as we are confident that "Setting{}.Get()" will return "", then I think we can resolve this.